web/text_layer_builder.js

```diff
@@ -83,13 +88,31 @@ class TextLayerBuilder {
    * @param {number} [timeout] - Wait for a specified amount of milliseconds
    *                             before rendering.
    */
-  render(timeout = 0) {
-    if (!(this.textContent || this.textContentStream) || this.renderingDone) {
-      return;
+  async render(viewport, timeout = 0) {
+    if (!(this.textContent || this.textContentStream)) {
+      return null;
```

Comment:
Shouldn't we throw an error in this case, since it's an implementation bug if we try and call `render` without either of these properties having been defined?
```suggestion
      throw new Error('Neither "textContent" nor "textContentStream" specified.');
```